Document new cases.* event-driven triggers (9.5) (#6480)
elastic/docs-content: explore-analyze/workflows/triggers/event-driven-triggers.md

Reviewer: ```suggestion
- [Triggers overview](/explore-analyze/workflows/triggers.md): All trigger types.
```

applies_to:
  stack: preview 9.4+
  serverless: preview

> description: Run a workflow in response to a platform event. Includes workflows.failed (added in 9.4) and the cases trigger family (added in 9.5).

Reviewer: I'm not actually sure whether we need to include versioning details in the description metadata. 

```suggestion
description: Run a workflow in response to a platform event. Includes workflows.failed and the cases trigger family.
```

# Event-driven triggers [workflows-event-driven-triggers]

Event-driven triggers let workflows react to events elsewhere in {{kib}}. Two trigger families are available:

> - **`workflows.failed`** — fires when another workflow's execution fails. Added in 9.4.

Reviewer: ```suggestion
- **`workflows.failed`** — Fires when another workflow's execution fails. {applies_to}`stack: ga 9.4+`
```

> - **Cases triggers** — fire when cases change (created, updated, status changed, attachments added, comments added). Added in 9.5.

Reviewer: ```suggestion
- **Cases triggers** — Fires when cases change (created, updated, status changed, attachments added, comments added). {applies_to}`stack: preview 9.5+` {applies_to}`serverless: preview`
```

> More event-driven triggers are planned for subsequent releases.

Reviewer: Optional edit. This is likely something users already expect.

```suggestion
```

:::{warning}

> The event-driven trigger system, including all triggers documented on this page, is in technical preview. The schema and semantics can change in future releases.

Reviewer: ```suggestion
The event-driven trigger system is in technical preview, including the triggers documented on this page. The schema and semantics can change in future releases.
```

      tags: ["workflow-failure", "auto-triage"]
```

## Cases triggers

```{applies_to}
stack: preview 9.5+
serverless: preview
```

Cases triggers fire when cases change. Use them to react to case lifecycle events without polling the Cases API.

**Shared payload.** Every cases trigger event includes:

> - `event.caseId` — the case ID.

Reviewer: Suggesting a bit of elaboration since there's [two types](https://www.elastic.co/docs/explore-analyze/cases/search-share-cases#case-identifiers) of numerical identifiers for cases. 

```suggestion
- `event.caseId` — The case ID, the alphanumeric identifier that is unique to each case.
```

> - `event.owner` — the case owner. One of `securitySolution`, `observability`, or `cases` (Stack Cases).

Reviewer: ```suggestion
- `event.owner` — The solution that owns the case. It can be `securitySolution` for {{elastic-sec}} cases, `observability` for Observability cases, or `cases` for Stack cases.
```

Use `event.owner` in `on.condition` to filter by solution. For example, a workflow that only fires for {{elastic-sec}} cases:

```yaml
triggers:
  - type: cases.caseCreated
    on:
      condition: 'event.owner: "securitySolution"'
```

Individual trigger sections below document any additional payload fields specific to that event.

### `cases.caseCreated` [cases-casecreated-trigger]

Fires when a case is created.

#### Schema

| Parameter | Location | Type | Required | Description |
|---|---|---|---|---|
| `type` | top level | string | Yes | Must be `cases.caseCreated`. |

> | `condition` | `on` | KQL string | No | Optional KQL predicate evaluated against the `event` payload. |

Reviewer: What is the `on` location? I feel like I might be missing something 😅

Author: Thanks for this question. It means the parameter lives inside an `on:` block. This value appears in other places in the workflows docs too. Added some info on line 166.

#### Event payload

| Field | Contains |
|---|---|
| `event.caseId` | The new case's ID. |
| `event.owner` | The case owner (`securitySolution`, `observability`, or `cases`). |

#### Example

Fire only for {{elastic-sec}} cases:

```yaml
triggers:
  - type: cases.caseCreated
    on:
      condition: 'event.owner: "securitySolution"'
```

### `cases.caseUpdated` [cases-caseupdated-trigger]

Fires when a case is updated. The `event.updatedFields` array lists which fields changed.

#### Schema

| Parameter | Location | Type | Required | Description |
|---|---|---|---|---|
| `type` | top level | string | Yes | Must be `cases.caseUpdated`. |
| `condition` | `on` | KQL string | No | Optional KQL predicate evaluated against the `event` payload. |

#### Event payload

| Field | Contains |
|---|---|
| `event.caseId` | The updated case's ID. |

> | `event.owner` | The case owner. |

Reviewer: To keep it consistent with what's doc'd above. Also helpful for users who directly land on this section from search instead of reading this page top-to-bottom. 

```suggestion
| `event.owner` | The case owner (`securitySolution`, `observability`, or `cases`). |
```

| `event.updatedFields` | Array of field names that changed in this update. |

#### Example

Fire when a {{elastic-sec}} case's title changes:

```yaml
triggers:
  - type: cases.caseUpdated
    on:
      condition: 'event.owner: "securitySolution" and event.updatedFields: "title"'
```

> % Ben Ironside Goldstein, 2026-05-13: SME flag — confirm with engineering whether

Reviewer: Just flagging as a reminder to remove if you've already confirmed this.

% `cases.caseUpdated` also fires when status changes (given the dedicated
% `cases.caseStatusUpdated` trigger), and how it behaves on bulk updates (one event
% per case, or one per batch).

### `cases.caseStatusUpdated` [cases-casestatusupdated-trigger]

Fires when a case's status changes.

#### Schema

| Parameter | Location | Type | Required | Description |
|---|---|---|---|---|
| `type` | top level | string | Yes | Must be `cases.caseStatusUpdated`. |
| `condition` | `on` | KQL string | No | Optional KQL predicate evaluated against the `event` payload. |

#### Event payload

| Field | Contains |
|---|---|
| `event.caseId` | The case ID. |

> | `event.owner` | The case owner. |

Reviewer: ```suggestion
| `event.owner` | The case owner (`securitySolution`, `observability`, or `cases`). |
```

| `event.previousStatus` | The previous status (`open`, `in-progress`, or `closed`). |
| `event.status` | The current status (`open`, `in-progress`, or `closed`). |

#### Example

Fire when a {{elastic-sec}} case is closed:

```yaml
triggers:
  - type: cases.caseStatusUpdated
    on:
      condition: 'event.owner: "securitySolution" and event.status: "closed"'
```

### `cases.attachmentsAdded` [cases-attachmentsadded-trigger]

Fires when attachments are added to a case. If attachments of multiple types are added in one operation (for example, three alerts and two comments), the trigger fires once per type, with one event for each type.

#### Schema

| Parameter | Location | Type | Required | Description |
|---|---|---|---|---|
| `type` | top level | string | Yes | Must be `cases.attachmentsAdded`. |
| `condition` | `on` | KQL string | No | Optional KQL predicate evaluated against the `event` payload. |

#### Event payload

| Field | Contains |
|---|---|
| `event.caseId` | The case ID. |

> | `event.owner` | The case owner. |

Reviewer: ```suggestion
| `event.owner` | The case owner (`securitySolution`, `observability`, or `cases`). |
```

| `event.attachmentType` | The type of attachments added, for example `"comment"` or `"alert"`. |

#### Examples

Fire only for {{elastic-sec}} cases:

```yaml
triggers:
  - type: cases.attachmentsAdded
    on:
      condition: 'event.owner: "securitySolution"'
```

Fire only when a comment-type attachment is added:

```yaml
triggers:
  - type: cases.attachmentsAdded
    on:
      condition: 'event.attachmentType: "comment"'
```

% Ben Ironside Goldstein, 2026-05-13: SME flag — confirm relationship between
% `cases.attachmentsAdded` (which can carry `attachmentType: "comment"`) and the
% dedicated `cases.commentsAdded` trigger. Do both fire when a comment is added,
% or is one a superset?

### `cases.commentsAdded` [cases-commentsadded-trigger]

Fires when comments are added to a case.

#### Schema

| Parameter | Location | Type | Required | Description |
|---|---|---|---|---|
| `type` | top level | string | Yes | Must be `cases.commentsAdded`. |
| `condition` | `on` | KQL string | No | Optional KQL predicate evaluated against the `event` payload. |

#### Event payload

| Field | Contains |
|---|---|
| `event.caseId` | The case ID. |

> | `event.owner` | The case owner. |

Reviewer: ```suggestion
| `event.owner` | The case owner (`securitySolution`, `observability`, or `cases`). |
```
